inst/apps/317-bslib-preset-shiny-dashboard/tests/testthat/test-317-bslib-preset-shiny-dashboard.R

```diff
@@ -0,0 +1,193 @@
+library(shinytest2)
+library(glue)
+
+# R Version to limit variants ---------------------------------
+resolve_r_version <- function(name) {
+  rver <- jsonlite::fromJSON(
+    glue("https://api.r-hub.io/rversions/resolve/{name}")
+  )
+  paste0(
+    strsplit(rver$version, ".", fixed = TRUE)[[1]][1:2],
+    collapse = "."
+  )
+}
+
+r_release <- resolve_r_version("release")
+
+if (platform_variant() != paste0("mac-", r_release)) {
+  skip("Screenshots only on MacOS with R-release")
+}
+
+this_platform <- platform_variant()
+this_platform <- sub(r_release, "release", this_platform, fixed = TRUE)
+
+# Setup App  --------------------------------------------------
+app <- AppDriver$new(
+  name = "317-bslib-preset-shiny-dashboard",
+  variant = this_platform,
+  height = 800,
+  width = 1200,
+  seed = 2023*11*13,
+  view = interactive(),
+  options = list(bslib.precompiled = FALSE),
+  screenshot_args = list(
+    selector = "viewport",
+    delay = 1,
+    options = list(captureBeyondViewport = FALSE)
+  )
+)
+withr::defer(app$stop())
+
+shinytest2_js <- local({
+  js_file <- system.file("internal", "js", "shiny-tracer.js", package = "shinytest2")
+  js_content <- readLines(js_file)
+  paste(js_content, collapse = "\n")
+})
+
+nav_to_variant <- function(app, ...) {
+  params <- list(...)
+  params <- purrr::compact(params)
+  params <- purrr::imap(params, function(value, name) sprintf("%s=%s", name, value))
+  params <- paste0(params, collapse = "&")
+
+  url <- sprintf("%s?%s", app$get_url(), params)
+  chrm <- app$get_chromote_session()
+
+  p <- chrm$Page$navigate(url, wait_ = FALSE)$
+    then(function(...) chrm$Page$loadEventFired(wait_ = FALSE))$
+    then(function(...) chrm$Runtime$evaluate(shinytest2_js, wait_ = FALSE))$
+    then(function(...) app$wait_for_idle())
```

Comment:
Might need to set up the load event as a promise, chaining off that action, rather than chaining from the navigate call. See https://github.com/rstudio/chromote/issues/102

```suggestion
  p <- chrm$Page$loadEventFired(wait_ = FALSE)$
    then(function(...) chrm$Runtime$evaluate(shinytest2_js, wait_ = FALSE))$
    then(function(...) app$wait_for_idle())
    
  chrm$Page$navigate(url, wait_ = FALSE)
```